### ISSUE TITLE
Not working on iOS

### ISSUE THREAD

**Issue by erperejildo** [ukasz123/soundpool#88]
Probably I'm doing something wrong but I don't know what.

```
import 'package:flutter/foundation.dart';
import 'package:flutter/services.dart';
import 'package:soundpool/soundpool.dart';

final mySounds = ['ball', 'good', 'remove', 'remove2', 'time'];

class Sounds extends ChangeNotifier {
  final list = {};
  Soundpool? pool;
  SoundpoolOptions soundpoolOptions = SoundpoolOptions();

  Sounds();

  void init() {
    if (this.list.isNotEmpty) {
      return;
    }
    pool?.dispose();
    pool = Soundpool.fromOptions(options: soundpoolOptions);
    print('pool updated: $pool');
    _load();
    notifyListeners();
  }

  Future _load() async {
    for (var i = 0; i < mySounds.length; i++) {
      var asset = await rootBundle.load("assets/sounds/${mySounds[i]}.mp3");
      int soundId = await pool!.load(asset);
      this.list[mySounds[i]] = soundId;
    }
  }

  Future<void> play(String sound) async {
    await pool!.play(list[sound]);
  }
}
```

On Android works fine but on iOS apart that I don't hear any sound, some animation that I'm running in the background stops. I don't get any error though

**Comment by erperejildo** [comment]
Tried on real device and this worked. Not sure why the simulator complains about this

**Comment by calcitem** [comment]
Have you resolved the issue of audio playback not working on iOS devices? I encountered a peculiar problem where the issue manifested on an iPhone Xs running iOS 17.4, but audio playback functioned normally on an iPad and in the iOS 17.4 simulator. When the iPhone Xs failed to produce sound, no abnormal log information was found in the console. Subsequently, I switched to using the [kplayer](https://pub.dev/packages/kplayer) package, and the iPhone Xs was able to generate audio as expected.
